packages/@aws-cdk/aws-apigateway/lib/integrations/async-proxy-handler/index.ts

```diff
@@ -0,0 +1,22 @@
+// tslint:disable:no-console
+import AWS = require('aws-sdk');
+
+export async function handler(event: any): Promise<AWSLambda.APIGatewayProxyResult> {
+  try {
+    if (!process.env.TARGET_FUNCTION_NAME) {
```

Comment:
This means that the handler can't be reused to serve multiple functions. That's okay but then why is it defined as a singleton? Is it possible perhaps to pass the function name through an http header?